src/main/java/janggi/board/Board.java

```diff
@@ -0,0 +1,114 @@
+package janggi.board;
+
+import janggi.Team.Team;
+import janggi.piece.Piece;
+import janggi.piece.PieceType;
+import janggi.position.Path;
+import janggi.position.Position;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Board {
+
+    private final Map<Position, Piece> pieces;
+
+    public Board(Map<Position, Piece> pieces) {
+        this.pieces = new HashMap<>(pieces);
+    }
+
+    public void move(final List<Position> positions, final Team currentTeam) {
+        Position currentPosition = positions.getFirst();
+        Position arrivalPosition = positions.getLast();
+        validateSamePosition(currentPosition, arrivalPosition);
+
+        Piece piece = findOwnPiece(currentTeam, currentPosition);
+        Path path = piece.makePath(currentPosition, arrivalPosition);
+
+        piece.validateExistPieceInPath(getPiecesByPath(path), hasPiece(arrivalPosition));
+        movePiece(piece, currentPosition, arrivalPosition, path);
+    }
+
+    public boolean hasEachKing() {
+        return calculateExistKing() == 2;
+    }
+
+    public Team findWinningTeam() {
+        if (calculateExistKing() != 1) {
```

Comment:
아래와 같이 표현하는것도 고려해볼 수 있겠어요!

```suggestion
        if (!hasEachKing()) {
```